src/main/java/org/jenkinsci/test/acceptance/WebDriverProvider.java

```diff
@@ -0,0 +1,209 @@
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.jenkinsci.test.acceptance;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Provider;
+
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.test.acceptance.guice.TestCleaner;
+import org.jenkinsci.test.acceptance.guice.TestName;
+import org.jenkinsci.test.acceptance.guice.TestScope;
+import org.jenkinsci.test.acceptance.selenium.SanityChecker;
+import org.jenkinsci.test.acceptance.selenium.Scroller;
+import org.jenkinsci.test.acceptance.utils.ElasticTime;
+import org.jenkinsci.test.acceptance.utils.SauceLabsConnection;
+import org.junit.runners.model.Statement;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.UnsupportedCommandException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.firefox.FirefoxProfile;
+import org.openqa.selenium.htmlunit.HtmlUnitDriver;
+import org.openqa.selenium.ie.InternetExplorerDriver;
+import org.openqa.selenium.phantomjs.PhantomJSDriver;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
+import org.openqa.selenium.remote.UnreachableBrowserException;
+import org.openqa.selenium.safari.SafariDriver;
+import org.openqa.selenium.support.events.EventFiringWebDriver;
+
+import com.google.inject.Inject;
+
+/**
+ * Provider to create a WebDriver instance, allowing injection of Firefox profile
+ * 
+ * @author Carlos Sanchez
+ *
+ */
+public class WebDriverProvider implements Provider<WebDriver>, com.google.inject.Provider<WebDriver> {
+
+    /** Browser property to set the default locale. */
+    private static final String LANGUAGE_SELECTOR = "intl.accept_languages";
+
+    /**
+     * PhantomJS browser property to set the default locale.
+     */
+    private static final String LANGUAGE_SELECTOR_PHANTOMJS = "phantomjs.page.customHeaders.Accept-Language";
+
+    @Inject
+    private TestCleaner cleaner;
+
+    @Inject
+    private TestName testName;
+
+    @Inject
+    private ElasticTime time;
+
+    @Inject
+    private FirefoxProfile firefoxProfile;
```

Comment:
something similar can be done for other browsers (`ChromeOptions`, `DesiredCapabilities`,...)

Without a way to customize the FirefoxProfile there is no way for tests to change the behavior of the browser. In this particular case a script taking more than the default 30s makes the test fail.